iroh-quinn/src/connection.rs

```diff
@@ -810,12 +815,24 @@ impl std::ops::Deref for ConnectionRef {
     }
 }
 
+/// Inner connection, use with care.
 #[derive(Debug)]
-pub(crate) struct ConnectionInner {
+pub struct ConnectionInner {
```

Comment:
do you really need the whole thing, or could you just have sth like `WeakConnectionHandle(Weak<ConnectionInner>)` that implements`reset_congestion_state`?

Reply:
That's a good idea, I adopted this.